routes/api.js

```diff
@@ -238,58 +240,99 @@ var removeSession = function(sessionId) {
     return delete sessions[sessionId];
 };
 
-var proxyRoute = function(req, res, next) {
-    proxy.web(req, res);
+// Create a kernel. Does not use the proxy instance because it must parse the
+// and reserialize the request body with additional information in certain
+// configurations.
+router.post('/kernels', bodyParser.json({ type: 'text/plain' }), function(req, res) {
```

Comment:
A little apprehensive about using the Express router within this file. I like keeping that just to *app.js*.

We could do a check in `proxyRoute` to see if the `req` is for `/api/kernels` and if so, do this block of code. See deleted block in `proxy.on('proxyRes', ...`